File: spotsense/src/main/java/com/spotsense/utils/SpotSenseNetworkUtils.java
package com.spotsense.utils;

import android.annotation.SuppressLint;
import android.content.Context;
import android.net.ConnectivityManager;

public final class SpotSenseNetworkUtils {

    public static boolean isConnected(Context context) {
        if (isInternetAvailable(context)) {
            return true;
        } else {
            SpotSenseAlertDialogUtils.showInternetAlert(context);
            return false;
        }
    }

    public static boolean isNetworkConnected(Context context) {
        return isInternetAvailable(context);
    }

    @SuppressLint("MissingPermission")
    private static boolean isInternetAvailable(Context context) {
        if (context != null) {
            ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            return connectivityManager.getActiveNetworkInfo() != null && connectivityManager.getActiveNetworkInfo().isConnectedOrConnecting();
        } else {
            return false;
        }
    }
}

File: spotsense/src/main/java/com/app/spotsense/utils/SpotSenseNetworkUtils.kt
package com.app.spotsense.utils

import android.annotation.SuppressLint
import android.content.Context
import android.net.ConnectivityManager
import com.app.spotsense.utils.SpotSenseAlertDialogUtils.showInternetAlert

object SpotSenseNetworkUtils {
    fun isConnected(context: Context?): Boolean {
        return if (isInternetAvailable(context)) true else {
            showInternetAlert(context)
            false
        }
    }

    fun isNetworkConnected(context: Context?): Boolean {
        return isInternetAvailable(context)
    }

    @SuppressLint("MissingPermission")
    private fun isInternetAvailable(context: Context?): Boolean {
        return if (context != null) {
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            connectivityManager.activeNetworkInfo != null && connectivityManager.activeNetworkInfo!!.isConnectedOrConnecting
        } else false
    }
}